install_files/ansible-base/roles/noble-migration/tasks/main.yml

```
@@ -0,0 +1,92 @@
+---
+
+- name: Check migration JSON on mon server
+  ansible.builtin.slurp:
+    src: /etc/securedrop-noble-migration-state.json
+  register: migration_json
+  ignore_errors: yes
+
+- name: Skip migration if already done
+  set_fact:
+    already_finished: "{{ not migration_json.failed and (migration_json.content | b64decode | from_json)['finished'] == 'Done' }}"
```

Could the JSON check ever succeed if `migration.json_failed == True`?

```suggestion
    already_finished: "{{ (migration_json.content | b64decode | from_json)['finished'] == 'Done' }}"
```

---

Noting for my own understanding:  We don't just `set_fact` with the value of `finished` here because later we'll use `ansible.builtin.wait_for` to wait/block until the server reaches the target state.

---

I believe we still need to verify the `ansible.builtin.slurp` step didn't fail, because it has ignore_errors: yes. Otherwise accessing `migration_json.content` will error.